pkg/tests/apis/provisioning/provisioning_test.go

```diff
@@ -314,31 +342,106 @@ func TestIntegrationProvisioning(t *testing.T) {
 	})
 
 	t.Run("creating repository creates folder", func(t *testing.T) {
-		// Just make sure the folder doesn't exist in advance.
-		err := folderClient.Resource.Delete(ctx, "thisisafolderref", metav1.DeleteOptions{})
-		if err != nil && !errors.IsNotFound(err) {
-			require.NoError(t, err, "deletion should either be OK or fail with NotFound")
-		}
+		cleanSlate(t)
 
-		_, err = client.Resource.Update(ctx,
+		_, err := client.Resource.Update(ctx,
 			helper.LoadYAMLOrJSONFile("testdata/github-example.yaml"),
 			metav1.UpdateOptions{},
 		)
 		require.NoError(t, err)
 
-		resp, err := folderClient.Resource.Get(ctx, "thisisafolderref", metav1.GetOptions{})
+		require.EventuallyWithT(t, func(collect *assert.CollectT) {
+			resp, err := folderClient.Resource.Get(ctx, "thisisafolderref", metav1.GetOptions{})
+			require.NoError(collect, err)
+			require.Equal(collect, "thisisafolderref", mustNestedString(resp.Object, "metadata", "name"))
+		}, time.Second*2, time.Millisecond*20)
+	})
+
+	t.Run("creating GitHub repository syncs from branch selected", func(t *testing.T) {
+		cleanSlate(t)
+
+		var githubClient *github.MockClient
+		helper.GetEnv().GitHubMockFactory.Constructor = func(ttc github.TestingTWithCleanup) github.Client {
+			if githubClient != nil {
+				return githubClient
+			}
+			githubClient = github.NewMockClient(ttc)
+
+			isCtx := mock.Anything // mock.IsType(context.Background())
```

Comment:
I want some thoughts here: `mock.IsType` seemingly doesn't work, so I just set it to `mock.Anything` for now. The actual q is: do we care about this type of test?